client/src/components/Post/Post.tsx

```diff
@@ -122,16 +119,35 @@ class Post extends Component<Props, State> {
         }
     }
 
-
     handleCancel = () => {
         this.setState({showModal: false})
     }
 
+    setIsShown= (index:number ) => {
+        this.setState({showDeleteComment:!this.state.showDeleteComment, hoverIndex:index})
+    }
+
+    handleCancelDeleteComment = () => {
+        this.setState({showDeleteCommentModal: false})
+    }
+
     handleConfirm = () => {
         deletePost(this.props.location.state ? this.props.location.state.post.id : this.props.match.params.id)
             .then(() => this.handleTopicNavigation())
     }
 
+    handleConfirmDeleteComment = () => {
+       deleteComment(this.state.commentToDeleteId)
+            .then(() => {
+                const newComments = [...this.state.comments]
+                console.log(newComments.length)
+                newComments.filter(comment=>comment.id!==this.state.commentToDeleteId)
+                this.setState( { comments: [...this.state.comments].filter(comment => comment.id !== this.state.commentToDeleteId), showDeleteCommentModal:false } )
+                console.log(newComments.length)
```

Comment:
@sofiasdz remove unused lines
- Console logs
- const new comments
- newComments.filter()...

Only keep the `set state`
Try always to make an effort to reduce unused/debug code from production code, it might take some time to see the code all over again, but, from experience, it is worth the effort :)